src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -266,235 +214,88 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InvertIf
-            cancellationToken As CancellationToken) As Boolean
-
-            Dim value = semanticModel.GetConstantValue(numericLiteralExpression)
-            If value.HasValue AndAlso TypeOf value.Value Is Integer AndAlso DirectCast(value.Value, Integer) = 0 Then
 
-                Dim symbol = semanticModel.GetSymbolInfo(variableExpression, cancellationToken).Symbol
 
-                If symbol IsNot Nothing AndAlso symbol.Name = "Length" Then
-                    Dim containingType = symbol.ContainingType
-                    If containingType IsNot Nothing AndAlso
-                       (containingType.SpecialType = SpecialType.System_Array OrElse
-                        containingType.SpecialType = SpecialType.System_String) Then
-
-                        Return True
-                    End If
-                End If
-            End If
-
-            Return False
+        Friend Overrides Function GetInvertIfText() As String
+            Return VBFeaturesResources.Invert_If
         End Function
 
-        Private Function TryNegateBinaryComparisonExpression(
-            expression As ExpressionSyntax,
-            semanticModel As SemanticModel,
-            cancellationToken As CancellationToken,
-            ByRef result As ExpressionSyntax) As Boolean
-
-            Dim inverses As Tuple(Of SyntaxKind, SyntaxKind) = Nothing
-            If s_comparisonInversesMap.TryGetValue(expression.Kind, inverses) Then
-                Dim binaryExpression = DirectCast(expression, BinaryExpressionSyntax)
-                Dim expressionType = inverses.Item1
-                Dim operatorType = inverses.Item2
-
-                ' Special case negating Length > 0 to Length = 0 and 0 < Length to 0 == Length
-                ' for arrays and strings. We can do this because we know that Length cannot be
-                ' less than 0.
-                If CanSimplifyToLengthEqualsZeroExpression(binaryExpression, semanticModel, cancellationToken) Then
-                    expressionType = SyntaxKind.EqualsExpression
-                    operatorType = SyntaxKind.EqualsToken
-                End If
-
-                result = SyntaxFactory.BinaryExpression(
-                    expressionType,
-                    binaryExpression.Left.Parenthesize(),
-                    SyntaxFactory.Token(
-                        binaryExpression.OperatorToken.LeadingTrivia,
-                        operatorType,
-                        binaryExpression.OperatorToken.TrailingTrivia),
-                    binaryExpression.Right.Parenthesize())
-
-                result = result _
-                    .WithLeadingTrivia(binaryExpression.GetLeadingTrivia()) _
-                    .WithTrailingTrivia(binaryExpression.GetTrailingTrivia())
-
-                Return True
-            End If
-
-            Return False
+        Friend Overrides Function GetBinaryOperation(expressionNode As SyntaxNode, semanticModel As SemanticModel) As IOperation
```

Comment:
This method is not needed. You can invoke `semanticModel.GetOperation(expressionNode)` without the cast to `BinaryExpressionSyntax`.